integration/spark/src/main/common/java/io/openlineage/spark/agent/lifecycle/plan/CreateDataSourceTableAsSelectCommandVisitor.java

```diff
@@ -28,11 +29,15 @@ public List<OpenLineage.OutputDataset> apply(LogicalPlan x) {
     CreateDataSourceTableAsSelectCommand command = (CreateDataSourceTableAsSelectCommand) x;
     CatalogTable catalogTable = command.table();
 
+    StructType schema = catalogTable.schema();
+    if (schema.fields().length == 0 && command.query().schema().fields().length > 0) {
```

Comment:
Maybe add a simple comment on the reasoning behind checking the schema in `CatalogTable` and defaulting to the output schema from the query?

Reply:
Done